### PR TITLE
Preserve float64 SiLU second derivatives

### LINKED ISSUE #126632: tf.nn.silu loses a finite second-derivative signal for float64 input
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Mobile device

_No response_

### Python version

Python 3.13

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

`tf.nn.silu` returns zero for a second-derivative component whose correctly rounded `float64` reference value is finite, normal, and nonzero.

SiLU is `x * sigmoid(x)`, and its second derivative follows from differentiating this product. At `x = 40.410841513445845`, the high-precision reference rounded to float64 is approximately `-1.0820525268131133e-16`.

However, TensorFlow autodiff returns `0.0` for this second derivative. The forward value is finite and correctly represented; the discrepancy is in the second-order autodiff result.

Expected behavior?

The second derivative of `tf.nn.silu` at `x = 40.410841513445845` should be close to `-1.0820525268131133e-16`, not `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x_value = 40.410841513445845
expected = -1.0820525268131133e-16
rtol = 1e-6

x = tf.Variable(x_value, dtype=tf.float64)

with tf.GradientTape() as outer:
    with tf.GradientTape() as inner:
        y = tf.nn.silu(x)
    g1 = inner.gradient(y, x)

g2 = outer.gradient(g1, x)

actual = float(g2)
rel_err = abs(actual - expected) / abs(expected)

print(f"x: {x_value!r}")
print(f"forward value: {float(y)!r}")
print(f"first derivative: {float(g1)!r}")
print(f"tensorflow second derivative: {actual!r}")
print(f"expected second derivative: {expected!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.silu loses a finite second-derivative signal")
else:
    print("not reproduced")
```

### Relevant log output

```shell
x: 40.410841513445845
forward value: 40.410841513445845
tensorflow second derivative: 0.0
expected second derivative: -1.0820525268131133e-16
relative error: 1.000e+00
BUG REPRODUCED: tf.nn.silu loses a finite second-derivative signal
```

### PR DESCRIPTION
**Pull request by GodlyDonuts**
Fixes #126632.

The custom SiLU gradient currently obtains its sigmoid tail as `1 - sigmoid(logits)`. For sufficiently large positive float64 logits, that subtraction rounds to zero and removes a finite second derivative.

For float64 gradients, evaluate the smaller sigmoid tail directly from the non-positive logit and reconstruct the sigmoid value with a complement only where needed. Reuse that stable tail for both the activation and beta gradients. Other dtypes retain the existing path.

Add a graph/eager regression for the reported input and correctly rounded second derivative.

Validation:
- `python3 -m py_compile tensorflow/python/ops/nn_impl.py tensorflow/python/ops/nn_grad_test.py`
- Numerical evaluation of the stable derivative formula at the reported float64 input

### CODE REVIEW

**Comment by GodlyDonuts** [comment]
@gemini-code-assist /review

**Comment by GodlyDonuts** [comment]
huh is it down? It usually is really fast and automatic

**Review by dmiltr3 — APPROVED** [review]
LGTM, TAP tests passed.
